examples/cpp/flash_attn/flash_attn_cpu.hpp

```diff
@@ -6,110 +6,120 @@ __half max(__half a, __half b) { return a > b ? a : b; }
+                acc[i * kN + j] = exp(acc[i * kN + j] - cur_row_max[i]);
+            }
         }
-    }
 
-    // Compute reduce sum for each row and store into `cur_sums`.
-    for (int i = 0; i < kM; ++i) {
-        __half sum = 0.;
-        for (int j = 0; j < kN; ++j) {
-            sum += acc[i * kN + j];
+        // Compute reduce sum for each row and store into `cur_sums`.
+        for (int i = 0; i < kM; ++i) {
+            __half sum = 0.;
+            for (int j = 0; j < kN; ++j) {
+                sum += acc[i * kN + j];
+            }
+            cur_sums[i] = sum;
         }
-        cur_sums[i] = sum;
-    }
 
-    // Compare cur row max with prev row max and compute new row max
-    for (int i = 0; i < kM; ++i) {
-        new_row_max[i] = max(cur_row_max[i], prev_row_max[i]);
-    }
+        // Compare cur row max with prev row max and compute new row max
+        for (int i = 0; i < kM; ++i) {
+            new_row_max[i] = max(cur_row_max[i], prev_row_max[i]);
+        }
 
-    for (int i = 0; i < kM; ++i) {
-        // Compute remormalization factor for the previous block.
-        prev_norm_vec[i] = exp(prev_row_max[i] - new_row_max[i]);
+        for (int i = 0; i < kM; ++i) {
+            // Compute remormalization factor for the previous block.
+            prev_norm_vec[i] = exp(prev_row_max[i] - new_row_max[i]);
 
-        // Compute remormalization factor for the current block.
-        new_norm_vec[i] = exp(cur_row_max[i] - new_row_max[i]);
-    }
+            // Compute remormalization factor for the current block.
+            new_norm_vec[i] = exp(cur_row_max[i] - new_row_max[i]);
+        }
 
-    // Update normalization factor l(x).
-    for (int i = 0; i < kM; ++i) {
-        new_sums[i] =
-            prev_norm_vec[i] * prev_sums[i] + new_norm_vec[i] * cur_sums[i];
-    }
+        // Update normalization factor l(x).
+        for (int i = 0; i < kM; ++i) {
+            new_sums[i] =
+                prev_norm_vec[i] * prev_sums[i] + new_norm_vec[i] * cur_sums[i];
+        }
 
-    // Compute unnormalizatied attention score @ values
-    for (int i = 0; i < kM; ++i) {
-        for (int j = 0; j < kP; ++j) {
-            __half s = 0.;
-            for (int k = 0; k < kN; ++k) {
-                s += acc[i * kN + k] * V[k + kN * j];
+        // Compute unnormalizatied attention score @ values
+        for (int i = 0; i < kM; ++i) {
+            for (int j = 0; j < kP; ++j) {
+                __half s = 0.;
+                for (int k = 0; k < kN; ++k) {
+                    s += acc[i * kN + k] * v_[k + kN * j];
+                }
+                exp_values[i * kP + j] = s;
             }
-            exp_values[i * kP + j] = s;
         }
-    }
 
 #ifdef DEBUG
-    printf("Print exp_values: \n");
-    for (int i = 0; i < kM; ++i) {
-        for (int j = 0; j < kP; ++j) {
-            printf("%.3f ", __half2float(exp_values[i * kP + j]));
+        printf("Print exp_values: \n");
```

Comment:
I noticed that you deleted the debug printing for the flash attention kernel in this pr. Therefore, are these debug printing still necessary? do you prefer still keeping it?

Reply:
Follow the comments and I will add pretty printer in the future PRs.